File: Java/Arrays/Easy/TwoSum.java
// Question
// Given an array nums and a target value,
//  find two numbers such that they add up to the target. 
// Return their indices.

package Java.Arrays.Easy;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target){
        for (int i=0; i<nums.length; i++ ){
            for (int j = 0; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[]{i, j};
                }
            }
        }
        return nums;
    }
     public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 22;
        int[] result = twoSum(nums, target);
        if (result.length == 2) {
            System.out.println("\n\nIndices: " + result[0] + " and " + result[1]);
        } else {
            System.out.println("No pair found.");
        }
    }
}

File: Swift/Arrays/TwoSum.swift
import Foundation 

func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    for i in 0..<nums.count{
        for j in 0..<nums.count{
            if nums[i] + nums[j] == target {
                print("Elements: \(nums[i]) and \(nums[j])")
                return [i, j]
            }
        }
    }
    return []
}

let nums = [2,1,5,6,7,9]
let target = 12
let result = twoSum(nums, target)
if result.isEmpty{
    print("No pair found.")
}else {
    print("Indices: \(result[0]) and \(result[1])")
}